src/lib/server/editors/editor.ts

```diff
@@ -44,8 +46,9 @@ export abstract class PromptEditor {
 	 * @param instancePredictions instances and predictions associated with the prompt
 	 * @returns the modified prompt
 	 */
-	abstract apply(
+	abstract rewritePrompt(
 		prompt: Tables<'prompts'>,
+		targetSpans: number[],
```

Comment:
should this be a more complex data structure? e.g. tuples in the array or something? not sure

Reply:
changed to tuples but supabase doesn't get it, only does 1d arrays...